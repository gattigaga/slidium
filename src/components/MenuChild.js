import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import Icon from "components/Icon";

const Container = styled.div`
  width: 100%;
  padding: 8px 16px;
  box-sizing: border-box;
  cursor: ${({ isClickable }) => (isClickable ? "pointer" : "default")};
  position: relative;
  display: flex;
  align-items: center;
  user-select: none;

  &:hover {
    ${({ isClickable }) => isClickable && "background: #eee;"};
  }
`;

const IconContainer = styled.div`
  width: 32px;
  display: flex;
  position: relative;
`;

const StyledIcon = styled(Icon)`
  font-size: 20px;
  color: ${({ isClickable }) => (isClickable ? "#888" : "#bbb")};
  position: absolute;
  top: 50%;
  left: 0%;
  transform: translateY(-50%);
`;

const Label = styled.span`
  font-size: 12px;
  font-family: Roboto;
  color: ${({ isClickable }) => (isClickable ? "black" : "#bbb")};
`;

const MenuChild = ({ label, icon, isClickable, onClick }) => (
  <Container isClickable={isClickable} onClick={() => isClickable && onClick()}>
    <IconContainer>
      {icon && <StyledIcon name={icon} isClickable={isClickable} />}
    </IconContainer>
    <Label isClickable={isClickable}>{label}</Label>
  </Container>
);

MenuChild.propTypes = {
  icon: PropTypes.string,
  label: PropTypes.string,
  isClickable: PropTypes.bool,
  onClick: PropTypes.func
};

MenuChild.defaultProps = {
  label: "Menu",
  isClickable: true,
  onClick: () => {}
};

export default MenuChild;
