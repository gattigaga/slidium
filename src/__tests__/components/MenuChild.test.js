import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import MenuChild from "components/MenuChild";

let wrapper, onClick;

beforeEach(() => {
  onClick = jest.fn();
  wrapper = shallow(<MenuChild label="Open" icon="print" onClick={onClick} />);
});

describe("MenuChild", () => {
  it("renders as clickable", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("renders as non clickable", () => {
    wrapper.setProps({ isClickable: false });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("calls 'onClick'", () => {
    expect(onClick.mock.calls.length).toEqual(0);
    wrapper.simulate("click");
    expect(onClick.mock.calls.length).toEqual(1);
  });

  it("not calls 'onClick'", () => {
    wrapper.setProps({ isClickable: false });

    expect(onClick.mock.calls.length).toEqual(0);
    wrapper.simulate("click");
    expect(onClick.mock.calls.length).toEqual(0);
  });
});
