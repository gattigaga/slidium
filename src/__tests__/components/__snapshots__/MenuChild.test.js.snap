// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuChild renders as clickable 1`] = `
<styled.div
  isClickable={true}
  onClick={[Function]}
>
  <styled.div>
    <Styled(Icon)
      isClickable={true}
      name="print"
    />
  </styled.div>
  <styled.span
    isClickable={true}
  >
    Open
  </styled.span>
</styled.div>
`;

exports[`MenuChild renders as non clickable 1`] = `
<styled.div
  isClickable={false}
  onClick={[Function]}
>
  <styled.div>
    <Styled(Icon)
      isClickable={false}
      name="print"
    />
  </styled.div>
  <styled.span
    isClickable={false}
  >
    Open
  </styled.span>
</styled.div>
`;
